File: Day07/PlusOne.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        a = ''
        for i in digits:
            a += str(i)
        b = int(a) + 1
        l = []
        for j in str(b):
            l.append(int(j))
        return l

def main():
    sol = Solution()
    test_cases = [
        [1, 2, 3],
        [9],
        [4, 3, 2, 1],
        [0]
    ]
    for digits in test_cases:
        print(f"plusOne({digits}) = {sol.plusOne(digits)}")

if __name__ == "__main__":
    main()

File: Day07/PlusOne.java
import java.util.Scanner;

public class PlusOne {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i]++;
            digits[i] %= 10;
            if (digits[i] != 0) {
                return digits;
            }
        }
        int[] result = new int[digits.length + 1];
        result[0] = 1;
        return result;
    }

    public static void main(String[] args) {
        PlusOne sol = new PlusOne();
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the digits array: ");
        int n = scanner.nextInt();

        int[] digits = new int[n];
        System.out.println("Enter the digits:");
        for (int i = 0; i < n; i++) {
            digits[i] = scanner.nextInt();
        }
        
        int[] result = sol.plusOne(digits);

        System.out.print("Result after plus one: ");
        for (int digit : result) {
            System.out.print(digit + " ");
        }
        System.out.println();

        scanner.close();
    }
}
